### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by ShogunPanda**
* Use sonic-boom
* Makes destination files customizable
* Add complete code coverage.

### CODE REVIEW

File: src/profile.js [hunk 1]
```diff
@@ -1,24 +1,30 @@
 'use strict'
 
 const { Session } = require('inspector')
-const { openSync, closeSync, writeSync } = require('fs')
+const { writeFileSync } = require('fs')
```
Comment (mcollina) [inline src/profile.js:4]: please use an async version.

File: src/snapshot.js [hunk 2]
```diff
@@ -1,57 +1,65 @@
...
 
+  const writer = new SonicBoom({ dest: destination })
+
+  writer.on('error', err => {
+    callback(err)
+  })
+
+  writer.on('close', () => {
+    callback(error, destination)
+  })
```
Comment (mcollina) [inline src/snapshot.js:32]: you need to remove those listeners to handle the case of 'close' and 'error' are both emitted.

File: test/snapshot.spec.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  const generated = JSON.parse(readFileSync(destination, 'utf-8'))
+
+  t.true(validate(generated))
+
+  cleanup()
+})
+
+t.test('it correctly generates a snapshot using callbacks', t => {
+  tmpFile().then(({ path: destination, cleanup }) => {
```
Comment (mcollina) [inline test/snapshot.spec.js:23]: please use an async function and await instead.

File: test/profile.spec.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  t.true(validate(generated))
+
+  cleanup()
+})
+
+t.test('it correctly generates a profile using callbacks', t => {
+  Promise.all([
+    allocateMemoryFor(100),
+    tmpFile().then(({ path: destination, cleanup }) => {
+      generateHeapSamplingProfile({ destination, duration: 100 }, () => {
```
Comment (mcollina) [inline test/profile.spec.js:27]: mixing callbacks and promises is really ugly and can have bugs. Can you please detangle this?

File: src/profile.js (hunk 1 above)
Comment (ShogunPanda) [reply src/profile.js]: Done!

File: test/snapshot.spec.js (hunk 3 above)
Comment (ShogunPanda) [reply test/snapshot.spec.js]: Done!

File: src/snapshot.js (hunk 2 above)
Comment (ShogunPanda) [reply src/snapshot.js:45]: Handle in a equivalent way.

File: test/profile.spec.js (hunk 4 above)
Comment (ShogunPanda) [reply test/profile.spec.js]: Agree. Changed to a "nicer" version.

**Pull request merged**